packages/project-management-automation/lib/add-milestone.js

```diff
@@ -32,19 +32,18 @@ const isDuplicateValidationError = ( error ) =>
 /**
  * Assigns the correct milestone to PRs once merged.
  *
- * @param {Object} payload Pull request event payload, see https://developer.github.com/v3/activity/events/types/#pullrequestevent.
+ * @param {Object} payload Push event payload, see https://developer.github.com/v3/activity/events/types/#pushevent.
  * @param {Object} octokit Initialized Octokit REST client, see https://octokit.github.io/rest.js/.
  */
 async function addMilestone( payload, octokit ) {
-	if ( ! payload.pull_request.merged ) {
-		debug( 'add-milestone: Pull request is not merged. Aborting' );
+	if ( payload.ref !== 'refs/heads/master' ) {
+		debug( 'add-milestone: Commit is not to `master`. Aborting' );
 		return;
 	}
 
-	if ( payload.pull_request.base.ref !== 'master' ) {
-		debug(
-			'add-milestone: Pull request is not based on `master`. Aborting'
-		);
+	const [ , prNumber ] = payload.commits[ 0 ].message.match( /\(#(\d+)\)$/ );
```

Comment:
Will it work to match the parenthesized pull request number as always at the end of the message?

For example, looking at the webhook payload for 818eb80edccbba5445e0eadeaf9c440e22fe6a86, I see:

```json5
{
  "commits": [
    {
      "message": "Fix plain text selection/focus (#19964)\n\n* Upgrade react native to 0.62.0-rc.1\r\n\r\n* Fix focus on plain text for RN\r\n\r\nIt looks there was a type and we were checking the incorrect props.\r\n\r\n* Revert change on package.json\r\n\r\nCo-authored-by: Marko Savic <savicmarko1985@gmail.com>",
      // ...
    }
  ],
  // ...
}
```

```js
"Fix plain text selection/focus (#19964)\n\n* Upgrade react native to 0.62.0-rc.1\r\n\r\n* Fix focus on plain text for RN\r\n\r\nIt looks there was a type and we were checking the incorrect props.\r\n\r\n* Revert change on package.json\r\n\r\nCo-authored-by: Marko Savic <savicmarko1985@gmail.com>".match( /\(#(\d+)\)$/ )
// null
```

I think it might be enough to use the `m` (multiline) flag:

```suggestion
	const [ , prNumber ] = payload.commits[ 0 ].message.match( /\(#(\d+)\)$/m );
```

Reply:
Good catch! Descriptions are concatenated there.